cli/mender-artifact/debugfs.go

```diff
@@ -104,53 +105,104 @@ func debugfsReplaceFile(imageFile, newFile, image string) (err error) {
 	executeCommand(cmd, image)
 	// Write to the partition
 	cmd = fmt.Sprintf("cd %s\nwrite %s %s\nclose", filepath.Dir(imageFile), newFile, filepath.Base(imageFile))
-	return executeCommand(cmd, image)
+	_, err = executeCommand(cmd, image)
+	return err
+}
+
+func debugfsRemoveFileOrDir(imageFile, image string, recursive bool) (err error) {
+	// Check that the file or directory exists.
+	cmd := fmt.Sprintf("cd %s", filepath.Dir(imageFile))
+	if _, err = executeCommand(cmd, image); err != nil {
+		return err
+	}
+	isDir := filepath.Dir(imageFile) == filepath.Clean(imageFile)
+	var deleteCmd string
+	if isDir {
+		return debugfsRemoveDir(imageFile, image, recursive)
+	} else {
+		deleteCmd = "rm"
+	}
+	cmd = fmt.Sprintf("%s %s\nclose", deleteCmd, imageFile)
+	_, err = executeCommand(cmd, image)
+	return err
+}
+
+func debugfsRemoveDir(imageFile, image string, recursive bool) (err error) {
+	// Remove the `/` suffix if present, as debugfs does not play nice with it.
+	imageFile = strings.TrimSuffix(imageFile, "/")
```

Comment:
Suggest using `TrimRight` instead of `TrimSuffix`. Then it will behave correctly also if there are multiple slashes.

Reply:
Nice! Did not know about this one.